docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md

```diff
@@ -0,0 +1,132 @@
+
+## Predicted Entities
+
+
+
+{:.btn-box}
+[Live Demo](https://demo.johnsnowlabs.com/healthcare/CONTEXTUAL_SPELL_CHECKER/){:.button.button-orange}
+[Open in Colab](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp-workshop/blob/master/tutorials/Certification_Trainings/Healthcare/6.Clinical_Context_Spell_Checker.ipynb){:.button.button-orange.button-orange-trans.co.button-icon}
+[Download](https://s3.amazonaws.com/auxdata.johnsnowlabs.com/clinical/models/spellcheck_clinical_en_3.4.1_3.0_1649672133997.zip){:.button.button-orange.button-orange-trans.arr.button-icon}
+
+## How to use
+
+
+
+<div class="tabs-box" markdown="1">
+{% include programmingLanguageSelectScalaPythonNLU.html %}
+```python
+documentAssembler = DocumentAssembler()\
+  .setInputCol("text")\
+  .setOutputCol("document")
+
+tokenizer = Tokenizer()\
+      .setInputCols(["document"])\
+      .setOutputCol("token")\
+      .setContextChars(["*", "-", "“", "(", "[", "\n", ".","\"", "”", ",", "?", ")", "]", "!", ";", ":", "'s", "’s"])
+
+spellModel = ContextSpellCheckerModel\
+    .pretrained('spellcheck_clinical', 'en', 'clinical/models')\
+    .setInputCols("token")\
+    .setOutputCol("checked")
+
+pipeline = Pipeline(stages = [documentAssembler, tokenizer, spellModel])
+
+empty = spark.createDataFrame([[""]]).toDF("text")
+lp = LightPipeline(pipeline.fit(empty))
+
+example = ["Witth the hell of phisical terapy the patient was imbulated and on postoperative, the impatient tolerating a post curgical soft diet.",
+           "With paint wel controlled on orall pain medications, she was discharged too reihabilitation facilitay.",
+           "Abdomen is sort, nontender, and nonintended.",
+           "Patient not showing pain or any wealth problems.",
+           "No cute distress"]
+
+lp.annotate(example)
+```
+```scala
+val assembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+val tokenizer = new Tokenizer()\
+      .setInputCols(["document"])\
+      .setOutputCol("token")\
+      .setContextChars(["*", "-", "“", "(", "[", "\n", ".","\"", "”", ",", "?", ")", "]", "!", ";", ":", "'s", "’s"])
+
+ val spellChecker = ContextSpellCheckerModel.
+      pretrained("spellcheck_clinical", "en", "clinical/models").
+      setInputCols("token").
+      setOutputCol("checked")
+
+ val pipeline =  new Pipeline().setStages(Array(assembler, tokenizer, spellChecker))
+ val empty_df = spark.createDataFrame([[""]]).toDF("text")
```

Comment:
Is this the way to create and empty DF in scala? I don't think the use of brackets is correct for Scala arrays

Reply:
I took it from existing page. Yes you are right. It should be fixed. I will

Reply:
fixed